From 0f24eddbcc7d63e843f598ae0296cdfdf47d123f Mon Sep 17 00:00:00 2001
From: Chris Adams <christopher.adams@nokia.com>
Date: Thu, 22 Jul 2010 11:31:36 +1000
Subject: [PATCH] Don't use imageprovider.h in declarative plugin due to source break

This is basically the same as 35a460a533fdbbcb4bc1d2659aed37414fa3e985
but applied to the 1.0 branch.
---
 examples/qmlcontacts/contents/example.vcf        |    1 +
 plugins/declarative/contacts/contacts.pro        |    7 +++++--
 plugins/declarative/contacts/plugin.cpp          |    5 +++--
 plugins/declarative/contacts/qmlcontactmodel.cpp |    3 ++-
 4 files changed, 11 insertions(+), 5 deletions(-)

--- a/examples/qmlcontacts/contents/example.vcf
+++ b/examples/qmlcontacts/contents/example.vcf
@@ -69,6 +69,7 @@ VERSION:2.1
 N:Edie;David;
 FN:Mr. David Edie
 ORG:Example;
+PHOTO:http://qt.nokia.com/logo.png
 TITLE:Manager
 NOTE:
 TEL;WORK;VOICE:(07) 3245-2323
--- a/plugins/declarative/contacts/contacts.pro
+++ b/plugins/declarative/contacts/contacts.pro
@@ -31,16 +31,19 @@ qmldir.path +=  $$[QT_INSTALL_IMPORTS]/$
 
 # Input
 HEADERS += qmlcontactmodel.h \
-           imageprovider.h \
            qmlcontact.h \
            qmlcontactdetail.h \
            qmlcontactdetailfield.h
 
 SOURCES += plugin.cpp \
     qmlcontactmodel.cpp \
-    imageprovider.cpp \
     qmlcontact.cpp \
     qmlcontactdetail.cpp \
     qmlcontactdetailfield.cpp
 
+# Qt 4.7.0b2 and 4.7.0rc1 have a source break for declarativeimageprovider, so don't
+# compile them for now.
+# HEADERS += imageprovider.h
+# SOURCES += imageprovider.cpp
+
 INSTALLS += target qmldir
--- a/plugins/declarative/contacts/plugin.cpp
+++ b/plugins/declarative/contacts/plugin.cpp
@@ -47,7 +47,8 @@
 #include "qmlcontact.h"
 #include "qmlcontactdetail.h"
 #include "qmlcontactdetailfield.h"
-#include "imageprovider.h"
+// Qt 4.7.0b2 and rc1 have a source break - turn off for now
+// #include "imageprovider.h"
 
 QT_USE_NAMESPACE
 
@@ -67,7 +68,7 @@ public:
 
     void initializeEngine(QDeclarativeEngine *engine, const char *uri) {
         Q_UNUSED(uri);
-        engine->addImageProvider("thumbnail", new ContactThumbnailImageProvider);
+        //engine->addImageProvider("thumbnail", new ContactThumbnailImageProvider);//due to source break
     }
 };
 
--- a/plugins/declarative/contacts/qmlcontactmodel.cpp
+++ b/plugins/declarative/contacts/qmlcontactmodel.cpp
@@ -270,7 +270,8 @@ QVariant QMLContactModel::data(const QMo
             return c.localId();
         case AvatarRole:
             //Just let the imager provider deal with it
-            return QString("image://thumbnail/%1.%2").arg(manager()).arg(c.localId());
+            //return QString("image://thumbnail/%1.%2").arg(manager()).arg(c.localId()); //imageprovider.h
+            return c.detail<QContactAvatar>().imageUrl();
         case Qt::DecorationRole:
             {
                 QContactThumbnail t = c.detail<QContactThumbnail>();
